debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -164,12 +169,19 @@ public Instant timestampOfLsn(Lsn lsn) throws SQLException {
         if (lsn.getBinary() == null) {
             return null;
         }
+
+        Instant cachedInstant = lsnToInstantCache.get(lsn);
+        if (cachedInstant != null) {
+            return cachedInstant;
+        }
+
         return prepareQueryAndMap(query, statement -> {
             statement.setBytes(1, lsn.getBinary());
         }, singleResultMapper(rs -> {
             final Timestamp ts = rs.getTimestamp(1);
             final Instant ret = (ts == null) ? null : ts.toInstant();
             LOGGER.trace("Timestamp of lsn {} is {}", lsn, ret);
+            Optional.ofNullable(ret).ifPresent(v -> lsnToInstantCache.put(lsn, ret));
```

Comment:
usage of `Optional` here - is an extra memory allocation, let's replace it to `if` condition

Reply:
Done.